src/core/border_router/routing_manager.hpp

```diff
@@ -1025,6 +1130,15 @@ class RoutingManager : public InstanceLocator
 
     DiscoveredPrefixStaleTimer mDiscoveredPrefixStaleTimer;
     RoutingPolicyTimer         mRoutingPolicyTimer;
+
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_ACCEPT_PLATFORM_ND_ENABLE
+    bool                   mDhcp6PdEnabled;
+    bool                   mPdOmrPrefixPreferred;
+    TimeMilli              mPdPrefixPreferredTime;
+    TimeMilli              mPdPrefixValidTime;
+    Ip6::Prefix            mPdOmrPrefix;
+    PlatformOmrPrefixTimer mPdOmrPrefixTimer;
```

Comment:
The more detailed idea is:
- The new logic, for handling the lifetime of the prefix from PD should go into a new class (this PR) since I think it is large enough.
- The current `LocalOmrPrefix` will be the state of new `OmrPrefixManager`, then `DetermineLocalOmrPrefix` (added in this PR),  `DetermineFavoredOmrPrefix`, and the logic in `LocalOmrPrefix` can be moved to `OmrPrefixManager`.